website_and_docs/content/documentation/webdriver/waits.en.md

```diff
@@ -1,495 +1,206 @@
-el = WebDriverWait(driver, timeout=3).until(lambda d: d.find_element(By.TAG_NAME,"p"))
-assert el.text == "Hello from JavaScript!"
+{{< gh-codeblock path="examples/python/tests/waits/test_waits.py#L18-L21" >}}
   {{< /tab >}}
   {{< tab header="CSharp" >}}
-   using (var driver = new FirefoxDriver())
-    {
-        var foo = new WebDriverWait(driver, TimeSpan.FromSeconds(3))
-                        .Until(drv => drv.FindElement(By.Name("q")));
-        Debug.Assert(foo.Text.Equals("Hello from JavaScript!"));
-    }
+{{< gh-codeblock path="examples/dotnet/SeleniumDocs/Waits/WaitsTest.cs#L27-L31" >}}
   {{< /tab >}}
   {{< tab header="Ruby" >}}
-  driver.get 'file:///race_condition.html'
-  wait = Selenium::WebDriver::Wait.new(:timeout => 10)
-  ele = wait.until { driver.find_element(css: 'p')}
-  foo = ele.text
-  assert_match foo, 'Hello from JavaScript'
+{{< gh-codeblock path="examples/ruby/spec/waits/waits_spec.rb#L19-L22" >}}
   {{< /tab >}}
   {{< tab header="JavaScript" >}}
-let ele = await driver.wait(until.elementLocated(By.css('p')),10000);
-let foo = await ele.getText();
-assert(foo == "Hello from JavaScript");
+{{< badge-code >}}
   {{< /tab >}}
   {{< tab header="Kotlin" >}}
-driver.get("file:///race_condition.html")
-val ele = WebDriverWait(driver, Duration.ofSeconds(10))
-            .until(ExpectedConditions.presenceOfElementLocated(By.tagName("p")))
-assert(ele.text == "Hello from JavaScript!")
+{{< badge-code >}}
   {{< /tab >}}
 {{< /tabpane >}}
 
-In that example, we pass in an anonymous function
-(but we could also define it explicitly as we did earlier so it may be reused).
-The first and only argument that is passed to our condition
-is always a reference to our driver object, _WebDriver_.
-In a multi-threaded environment, you should be careful
-to operate on the driver reference passed in to the condition
-rather than the reference to the driver in the outer scope.
+## Implicit waits
+Selenium has a built-in way to automatically wait for elements called an _implicit wait_.
+You set an implicit wait either with the [timeouts](({{< ref "drivers/options#timeouts" >}}))
+capability in the browser options, or with a driver method (as shown below).
 
-Because the wait will swallow _no such element_ errors
-that are raised when the element is not found,
-the condition will retry until the element is found.
-Then it will take the return value, a _WebElement_,
-and pass it back through to our script.
+This is a global setting that applies to every element location call for the entire session.
+The default value is `0`, which means that if the element is not found, it will
+immediately return an error. If an implicit wait is set, the driver will wait for the 
+duration of the provided value before returning the error. Note that as soon as the 
+element is located, the driver will return the value and your code may continue, so a larger
+implicit wait value won't necessarily increase the time of your session.
```

Comment:
i would rephrase to say something like "increase the duration of the test execution"